test/shared/src/main/scala/zio/test/Trace.scala

```diff
@@ -197,7 +197,12 @@ object Trace {
   }
 
   private[test] case class Or(left: Trace[Boolean], right: Trace[Boolean]) extends Trace[Boolean] {
-    override def result: Result[Boolean] = left.result.zipWith(right.result)(_ || _)
+    override def result: Result[Boolean] =
+      (left.result, right.result) match {
+        case (Result.Succeed(true), _) => Result.succeed(true)
+        case (_, Result.Succeed(true)) => Result.succeed(true)
```

Comment:
I don't think we want this because it violates the ordinary meaning of `false || true`.